### PR TITLE
[system health daemon] Support PSU power threshold checking

### PR DESCRIPTION
**Pull request by stephenxs**
#### Why I did it

Support PSU power threshold checking in the system health daemon.

Signed-off-by: Stephen Sun <stephens@nvidia.com>

#### How I did it

The system health daemon will check whether PSU's `power_overload` is `True` and treat it as not health if it is.

#### How to verify it

Unit test and regression test.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205

### CODE REVIEW

**Review by Junchao-Mellanox — APPROVED** [review]

**Review by keboliu — APPROVED** [review]

File: src/system-health/health_checker/hardware_checker.py [hunk 1]
```diff
@@ -239,6 +239,18 @@ def _check_psu_status(self, config):
                                                                                                                voltage_min_th,
                                                                                                                voltage_max_th))
                         continue
+
+            if not self._ignore_check(config.ignore_devices, 'psu', name, 'power_threshold'):
+                power_overload = data_dict.get('power_overload', None)
+                if power_overload == 'True':
+                    try:
+                        power = data_dict['power']
+                        power_critical_threshold = data_dict['power_critical_threshold']
```
Comment (prgeor) [inline src/system-health/health_checker/hardware_checker.py:248]: what if 'power_critical_threshold' is not applicable or supported for a platform

Comment (stephenxs) [reply src/system-health/health_checker/hardware_checker.py:248]: there will be an `N/A` in the entry. it is pushed by the PSU daemon.

File: src/system-health/health_checker/hardware_checker.py [hunk 2]
```diff
@@ -239,6 +239,18 @@ def _check_psu_status(self, config):
...
+                        self.set_object_not_ok('PSU', name, 'power of {} ({}w) exceeds threshold ({}w)'.format(name, power, power_critical_threshold))
+                    except KeyError:
+                        self.set_object_not_ok('PSU', name, 'power of {} exceeds threshold but power or power_critical_threshold does not invalid'.format(name))
```
Comment (prgeor) [inline src/system-health/health_checker/hardware_checker.py:251]: 'does not' -> 'is not'

Comment (stephenxs) [reply src/system-health/health_checker/hardware_checker.py]: Fixed

Comment (stephenxs) [reply src/system-health/health_checker/hardware_checker.py]: fixed typo: `is not invalid` => `is invalid`.

**Review by prgeor — APPROVED** [review]

**Pull request merged**